parse.go

```diff
@@ -89,7 +89,11 @@ func Parse(
 
 	// Check that `htm` and `htu` claims match the HTTP method and URL of the current request.
 	// This satisfies point 8 and 9 in https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop#section-4.3
-	if httpMethod != claims.Method || httpURL.String() != claims.URL {
+	
+	// Addresses https://github.com/AxisCommunications/go-dpop/issues/9
+	httpUrlParsed := strings.Join([]string{httpURL.Scheme, "://", httpURL.Hostname(), httpURL.Path}, "")
```

Comment:
```suggestion
	httpUrlParsed := fmt.Sprintf("%s://%s%s", httpURL.Scheme, httpURL.Host, httpURL.Path)
```

Reply:
I'm keeping httpURL.Hostname() instead of host due to port number parsing and "SHOULD" recommendation for scheme normalization in the dpop spec https://datatracker.ietf.org/doc/html/rfc9449#name-checking-dpop-proofs . If there's a reason to not do this then httpURL.Host works.

Reply:
Good point. I won't complain.